server/repositories/WalletRepository.js

```diff
@@ -98,6 +98,9 @@ class WalletRepository extends BaseRepository {
 
     query = query.union(union1, union2);
 
+    // get the total count (before applying limit and offset options)
+    const count = await this._session.getDB().from(query.as('p')).count('*');
```

Comment:
if getCount is false, I don't want us making this extra call. We can set this query at this point without a limit attached to it to a variable and not call it, and then call it using await in the if(getCount) statement